include/zephyr/sys/util.h

```diff
@@ -685,6 +685,16 @@ char *utf8_trunc(char *utf8_str);
  */
 char *utf8_lcpy(char *dst, const char *src, size_t n);
 
+/**
+ * @brief Checks if the given string @p str is UTF-8 encode.
+ *
+ * @param str Target string
+ * @param len The length of string @p str
```

Comment:
What happens if there is a `\0` byte before the full string `len`?
There might need a strategy for handling this case.

Reply:
I default this user to process it to ensure that the implementation of the function is essential.

Reply:
If we want to treat all inputs as NULL terminated strings (which string functions in C typically do, see e.g. `strlen`), then I suggest to not have the `len` parameters. 

Alternatively we can keep the `len` parameter if we want to have them as more secure functions like `strnlen`, and rename the `len` to `maxlen`. This has the benefit of also allowing strings that aren't NULL terminated to be passed to the function

Reply:
I agree your point. I'll to change.